examples/org.eclipse.swt.examples/src/org/eclipse/swt/examples/controlexample/TreeTab.java

```diff
@@ -258,6 +262,64 @@ void createOtherGroup () {
 		resizableColumns.addSelectionListener (widgetSelectedAdapter(event -> setColumnsResizable ()));
 		headerImagesButton.addSelectionListener (widgetSelectedAdapter(event -> recreateExampleWidgets ()));
 		subImagesButton.addSelectionListener (widgetSelectedAdapter(event -> recreateExampleWidgets ()));
+		editableButton.addSelectionListener(widgetSelectedAdapter(event -> makeTreeContentEditable()));
+	}
+
+	void makeTreeContentEditable() {
+		// Tree 1
+		final TreeEditor editor = new TreeEditor(tree1);
+		editor.horizontalAlignment = SWT.LEFT;
+		editor.grabHorizontal = true;
+
+		tree1.addListener(SWT.MouseDoubleClick, event -> {
+			treeDoubleClickListener(editor, event);
+
+		});
+
+		// Tree 2
+		final TreeEditor editor2 = new TreeEditor(tree2);
+		editor2.horizontalAlignment = SWT.LEFT;
+		editor2.grabHorizontal = true;
+
+		tree2.addListener(SWT.MouseDoubleClick, event -> {
+			treeDoubleClickListener(editor2, event);
+		});
```

Comment:
Also, the configuration of the tree editor and adding the listener is repeated code, which could be combined in separate `makeTreeEditable(Tree)` method.